prov/opx/include/rdma/opx/fi_opx_endpoint.h

```diff
@@ -1278,26 +1367,29 @@ void complete_receive_operation_internal (struct fid_ep *ep,
 
 #ifndef NDEBUG
 			uint8_t *orig_recv_buf_end = ((uint8_t *) recv_buf) + context->len;
-			uint8_t *payload_end = ((uint8_t *)recv_buf) + send_len;
-			for (uint8_t *buf_check = (uint8_t *)recv_buf + hdr->mp_eager_nth.payload_offset;
-				buf_check < payload_end; ++buf_check) {
-
-				if (*buf_check != 0xAA) {
-					fprintf(stderr, "(%d) %s:%s:%d Multi-Packet Eager Nth packet encountered "
-						"corrupted destination buffer! Initial PSN = %u, recv_buf=%p, "
-						"offset=%0X, payload_length=%ld, buffer altered at %p, "
-						"current value is %0hhX\n",
-						getpid(), __FILE__, __func__, __LINE__,
-						hdr->mp_eager_nth.mp_egr_uid, recv_buf,
-						hdr->mp_eager_nth.payload_offset, send_len,
-						buf_check, *buf_check);
-					abort();
+			if (!is_hmem) {
+				uint8_t *payload_end = ((uint8_t *)recv_buf) + send_len;
+				for (uint8_t *buf_check = (uint8_t *)recv_buf + hdr->mp_eager_nth.payload_offset;
```

Comment:
For this comment and the next one where you said:

> the convention is not to interleave declarations with statements.

Is there a particular guideline about this? I didn't see anything in the Linux Kernel coding guidelines about it (perhaps I missed it)? I know that much older versions of the C standard did not support this, but I would think that the best practice of limiting variable scope to where it is needed would be preferable?

Reply:
Yeah it's not in the current Linux kernel guide: https://www.kernel.org/doc/Documentation/process/coding-style.rst, but we have been sticking to this in the past. The opx provider was not strictly reviewed previously, partially due the amount of changes presented for each PR. 

Mixing declaration and statement makes code harder to read.  The scope is only partially limited since the other end is open. To really serve the purpose of limiting scope, use `{` and `}` instead.

I will update our coding style page to have this one explicitly listed.